### ISSUE TITLE
numpy 1.10 breaks master

### ISSUE THREAD

**Issue by arokem** [nipy/dipy#730]
In particular, several errors of this nature: 

    TypeError: Cannot cast ufunc add output from dtype('float64') to dtype('uint16') with casting rule 'same_kind'

**Comment by arokem** [comment]
Here's a minimal example of what's going on: 

```
In [1]: import numpy as np

In [2]: a = np.array([1,2,3])

In [3]: b = 2.0

In [4]: a /= b
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-b9f4626d3ca8> in <module>()
----> 1 a /= b

TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''

In [5]: a = a/b

In [6]: a
Out[6]: array([ 0.5,  1. ,  1.5])

```

**Comment by arokem** [comment]
Does anyone have an idea if this is a bug or a feature?

**Comment by arokem** [comment]
Apparently it's a feature, see: https://github.com/numpy/numpy/issues/6464

**Comment by arokem** [comment]
Resolved through #731
